runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3178,15 +3178,51 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 {
 	MM_GCExtensions* extensions = MM_GCExtensions::getExtensions(vmThread);
 	J9JavaVM *vm = vmThread->javaVM;
-	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * but not the original heap geometry from the snapshot run. We just force the heap
+	 * not to go beyond a certain thereshold calculated based on the new (restore) memory
+	 * availalibity through setting SoftMx.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	if(!extensions->userSpecifiedParameters._Xmx._wasSpecified){
+		uintptr_t candidateSoftMx = 0;
+		if(0.0 >= extensions->maxRAMPercent){
+			/* We use false here for computeDefaultMaxHeapForJava(), since the restore
+			* path is only active for releases after Java 8 releases
+			*/
+			candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+		}
+		else{// if the user have set maxRAMPercent through -XX:MaxRAMPercentage
+			candidateSoftMx = extensions->maxRAMPercent * extensions->usablePhysicalMemory / 100.0;
+		}
```

Comment:
Multiple coding standards issues:
- "else" should be at the same line as closing }
- space between else and {
- comment should be located at separate line
- comment should have /* ... */ style, not //
```suggestion
		} else {
		    /* if the user have set maxRAMPercent through -XX:MaxRAMPercentage */
			candidateSoftMx = extensions->maxRAMPercent * extensions->usablePhysicalMemory / 100.0;
		}
```

Reply:
Thanks for pointing them out!